Tests/UnitTests/Mocks/Sessions+Services/Analytics/MockAnalyticsService.swift

```diff
@@ -37,7 +37,9 @@ final class MockAnalyticsService: AnalyticsService {
         parameters["AnalyticsParameterScreenClass"] = screen.type.name
         parameters["AnalyticsParameterScreenName"] = screen.name
         
-        screenParamsLogged = parameters
+        screenParamsLogged["title"] = parameters["title"]
```

Comment:
🤔 I'm not sure this change is right, we should track all the parameters, surely? Otherwise the test should fail!

Reply:
There's likely a more elegant approach, but before with the original thing we had, more parameters were being added, but the`ScreenView` only cares about the title and screen ID, if it's an error we also need the reason. 

This is the failing error I got at the time when I ran it pre-change if it helps explain the above/my thinking: 
`XCTAssertEqual failed: ("["AnalyticsParameterScreenName": "you've been signed out", "AnalyticsParameterScreenClass": "signOutWarningScreen", "screen_id": "signoutwarningscreen", "title": "you've been signed out"]") is not equal to ("["title": "you've been signed out", "screen_id": "signoutwarningscreen"]")`

Reply:
Here's the ScreenParameters inside the actual `ScreenView` struct for easy reference: 
```
    public var parameters: [String: String] {
        [
            ScreenParameter.id.rawValue: id,
            ScreenParameter.title.rawValue: title
        ].compactMapValues(\.?.formattedAsParameter)
    }
```

Reply:
If we have more parameters being added, then that sounds like the test is failing for good reason. I'd suggest we should look into the root cause of the extra parameters 🙂

Reply:
Maybe I misunderstood the error message, but weren't the extra params being added by the Mock (as it was adding all of the params), but the expected outcome in the real dictionary was `["title": "you've been signed out", "screen_id": "signoutwarningscreen"]`?

Reply:
hmm 👀 why are we adding extra parameters in the mock? That doesn't sound right 🤔 
The mock should just be for validating we're passing in the right parameters.

Reply:
I'm not sure why we were initially, but it would explain the original testing approach. My update to the mock should now be passing in only the 3 parameters the tests were failing against. I did raise a tech debt ticket for this yesterday, to update the rest of the tests so I could expand on what the ticket is for to investigate/update how the mock handles these parameters?